src/Sequence.cpp

```diff
@@ -17,6 +18,8 @@ Sequence::Sequence(std::shared_ptr<vk::PhysicalDevice> physicalDevice,
 
     this->createCommandPool();
     this->createCommandBuffer();
+    if(nrOfTimestamps>0)
+        this->createTimestampQueryPool(nrOfTimestamps+1); //+1 for the first one
```

Comment:
Why is this called on sequence and not on begin? Would we want a timestamp that stores multiple eval runs?

Reply:
Because command pool and command buffer are also created in the constructor. What would be the difference anyway?

Reply:
> Would we want a timestamp that stores multiple eval runs?

How can a timestamp store anything? The query pool stores timestamps and yes, it is updated after each `eval()`. Timing multiple `eval` runs is definitely useful to estimate the average time an operations takes.